app/common/view.c

```diff
@@ -231,14 +245,63 @@ zxerr_t h_review_update_data() {
         viewdata.pageIdx = 0;
         return zxerr_ok;
     }
+
+    if (h_paging_intro_screen()) {
+        char* intro_key = NULL;
+        char* intro_value = NULL;
+
+#if defined(REVIEW_SCREEN_ENABLED)
+        switch (viewdata.itemIdx) {
+            case 0:
+                intro_key = PIC(review_key);
+                switch (review_type)
+                {
+                case REVIEW_UI:
+                    intro_key = PIC(review_keyconfig);
+                    intro_value = PIC(review_configvalue);
+                    break;
+
+                case REVIEW_ADDRESS:
+                    intro_value = PIC(review_addrvalue);
+                    break;
+
+                case REVIEW_TXN:
+                default:
+                    intro_value = PIC(review_txvalue);
+                    break;
+                }
+                break;
+        #if defined(APP_BLIND_MODE_ENABLED)
+            case 1:
+                intro_key = PIC(blindsigning_key);
+                intro_value = PIC(blindsigning_value);
+                break;
+        #endif
+            default:
+                return zxerr_no_data;
+        }
+#elif defined(APP_BLIND_MODE_ENABLED)
+        intro_key = PIC(blindsigning_key);
+        intro_value = PIC(blindsigning_value);
+#else
+        return zxerr_no_data;
+#endif
+
+        snprintf(viewdata.key, MAX_CHARS_PER_KEY_LINE, "%s", intro_key);
+        snprintf(viewdata.value, MAX_CHARS_PER_VALUE1_LINE, "%s", intro_value);
+        splitValueField();
+        viewdata.pageIdx = 0;
+        return zxerr_ok;
+    }
 #endif
 
     do {
         CHECK_ZXERR(viewdata.viewfuncGetNumItems(&viewdata.itemCount))
+        viewdata.itemCount+= getIntroPages();
 
         //Verify how many chars fit in display (nanos)
         CHECK_ZXERR(viewdata.viewfuncGetItem(
-                viewdata.itemIdx,
+                viewdata.itemIdx - getIntroPages(),
```

Comment:
we need to be 100% sure that `getIntroPages` cannot go negative or very large.

Can you calculate and compare before (in line 301) `viewdata.itemIdx` and `getIntroPages()` to be sure it is impossible to substract invalid values?